lib/websocket/connection.js

```diff
@@ -280,71 +285,95 @@ class WebsocketFrame {
 
     // 4. set payload length
     // TODO: support payload lengths larger than 125
-    buffer[1] += payloadLength
+    buffer[1] += this.data.length
 
-    if (this.isMasked) {
+    if (this.maskKey) {
     // 6. mask payload data
-      const maskKey = buffer.slice(2, 6)
       /*
      j                   = i MOD 4
      transformed-octet-i = original-octet-i XOR masking-key-octet-j
     */
-      for (let i = 0; i < payloadLength; i++) {
-        buffer[6 + i] = this.payloadData[i] ^ maskKey[i % 4]
+      for (let i = 0; i < this.data.length; i++) {
+        buffer[6 + i] = this.data[i] ^ this.maskKey[i % 4]
       }
     }
 
     return buffer
   }
 
   mask () {
-    this.isMasked = true
     this.maskKey = Buffer.allocUnsafe(4)
 
     for (let i = 0; i < 4; i++) {
       this.maskKey[i] = Math.floor(Math.random() * 256)
     }
   }
 
-  static from (buffer) {
-    let nextByte = 0
-    const frame = new WebsocketFrame()
+  size () {
```

Comment:
This is the size when the frame is converted to a buffer. I don't like the name so any suggestions would be great

Reply:
maybe something like byteLength to match existing apis?